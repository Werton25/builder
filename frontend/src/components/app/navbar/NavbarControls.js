import React from "react";
import PropTypes from 'prop-types';
import {resetLocalStorage} from "../../../actions/requests";
import {connect} from 'react-redux';
import './Navbar.css';
import ImportButton from "../buttons/ImportButton";
import {saveJsonFile} from "../../../utils";

export class BaseNavbarControls extends React.Component {
  onExportAll = () => {
    saveJsonFile('requests', this.props.requests);
  };

  render() {
    const {resetLocalStorage} = this.props;

    return (
      <li>
        <div className='btn-group'>
          <label className='btn btn-primary'>
            <ImportButton/>
            Загрузить из файла
          </label>
          <button className='btn btn-primary' type='button' onClick={this.onExportAll}>
            Сохранить в файл
          </button>
        </div>
        <button className='btn btn-danger btn-margin-left' type='button' onClick={resetLocalStorage}>
          Сбросить всё
        </button>
      </li>
    );
  }
}

BaseNavbarControls.propTypes = {
  requests: PropTypes.object.isRequired,
  resetLocalStorage: PropTypes.func.isRequired,
};

const mapStateToProps = ({requests}) => ({requests});

const mapDispatchToProps = {
  resetLocalStorage
};

export default connect(mapStateToProps, mapDispatchToProps)(BaseNavbarControls);
