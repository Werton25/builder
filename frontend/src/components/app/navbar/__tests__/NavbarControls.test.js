import React from "react";
import renderer from 'react-test-renderer';
import NavbarControls from "../NavbarControls";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

const store = configureStore()({
  requests: {}
});

describe('NavbarControls tests', () => {
  it('NavbarControls renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <NavbarControls/>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
});
