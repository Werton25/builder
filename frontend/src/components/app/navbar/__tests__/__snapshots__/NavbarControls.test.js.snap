// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarControls tests NavbarControls renders correctly 1`] = `
<li>
  <div
    className="btn-group"
  >
    <label
      className="btn btn-primary"
    >
      <input
        className="hidden"
        onChange={[Function]}
        type="file"
      />
      Загрузить из файла
    </label>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Сохранить в файл
    </button>
  </div>
  <button
    className="btn btn-danger btn-margin-left"
    onClick={[Function]}
    type="button"
  >
    Сбросить всё
  </button>
</li>
`;
